app/models/container_replicator.rb

```diff
@@ -9,5 +9,19 @@ class ContainerReplicator < ActiveRecord::Base
   has_many :selector_parts, -> { where(:section => "selectors") }, :class_name => "CustomAttribute", :as => :resource, :dependent => :destroy
   has_many :container_nodes, -> { distinct }, :through => :container_groups
 
+  include EventMixin
+
   acts_as_miq_taggable
+
+  def event_where_clause(assoc = :ems_events)
+    case assoc.to_sym
+    when :ems_events
+      # TODO: improve relationship using the id
+      ["container_namespace = ? AND container_replicator_name = ? AND #{events_table_name(assoc)}.ems_id = ?",
+       container_project.name, name, ems_id]
```

Comment:
> shouldn't the last parameter be ext_management_system.id?

I maintained consistency with pods and nodes.
belongs_to  :ext_management_system, :foreign_key => "ems_id"